From 4982923fdc7be9978d3b089fb443bcb0e20243eb Mon Sep 17 00:00:00 2001
From: Alberto Mardegan <alberto.mardegan@canonical.com>
Date: Tue, 21 Jun 2016 13:35:03 +0300
Subject: [PATCH] Build: fix value of default options in configure.ac

GLib assertions where always disabled, even if the --disable-*
parameters where not passed to configure.ac.

Quick reference:
http://www.robster.org.uk/2011/02/07/autofoo-ac_arg_enable/
---
 configure.ac |  9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/configure.ac b/configure.ac
index 364f9f2..3cde223 100644
--- a/configure.ac
+++ b/configure.ac
@@ -63,17 +63,20 @@ AM_CONDITIONAL([ENABLE_TESTS], [test "x$have_check" = "xyes"])
 GTK_DOC_CHECK([1.14], [--flavour no-tmpl])
 
 AC_ARG_ENABLE([cast-checks],
-    [AS_HELP_STRING([--disable-cast-checks], [compile with GLib cast checks disabled])])
+    [AS_HELP_STRING([--disable-cast-checks], [compile with GLib cast checks disabled])],
+    [], [enable_cast_checks=yes])
 AS_IF([test "x$enable_cast_checks" = "xno"],
     [CFLAGS="$CFLAGS -DG_DISABLE_CAST_CHECKS"])
 
 AC_ARG_ENABLE([asserts],
-    [AS_HELP_STRING([--disable-asserts], [compile with GLib assertions disabled])])
+    [AS_HELP_STRING([--disable-asserts], [compile with GLib assertions disabled])],
+    [], [enable_asserts=yes])
 AS_IF([test "x$enable_asserts" = "xno"],
     [CFLAGS="$CFLAGS -DG_DISABLE_ASSERTS"])
 
 AC_ARG_ENABLE([checks],
-    [AS_HELP_STRING([--disable-checks], [compile with GLib checks disabled])])
+    [AS_HELP_STRING([--disable-checks], [compile with GLib checks disabled])],
+    [], [enable_checks=yes])
 AS_IF([test "x$checks" = "xno"],
     [CFLAGS="$CFLAGS -DG_DISABLE_CHECKS"])
 
--
libgit2 0.26.0

